internal/mining/mining.go

```diff
@@ -1285,9 +1408,59 @@ mempoolLoop:
 		// Store if this is an SSRtx or not.
 		isSSRtx := prioItem.txType == stake.TxTypeSSRtx
 
+		var isTAdd, isTSpend bool
+		if isTreasuryEnabled {
+			// Store if this is a TAdd or not.
+			isTAdd = prioItem.txType == stake.TxTypeTAdd
+
+			// Store if this is an TSpend or not.
+			isTSpend = prioItem.txType == stake.TxTypeTSpend
+		}
+
 		// Grab the list of transactions which depend on this one (if any).
 		deps := dependers[*tx.Hash()]
 
+		// Skip TSpend if this block is not on a TVI or outside of the
+		// Tspend window.
+		if isTSpend {
+			if !standalone.IsTreasuryVoteInterval(uint64(nextBlockHeight),
+				g.chainParams.TreasuryVoteInterval) {
+
+				log.Tracef("Skipping tspend %v because block is not on a "+
+					"TVI: %v", tx.Hash(), nextBlockHeight)
+				continue
+			}
+
+			// We are on a TVI, make sure this tspend is within the
+			// window.
+			exp := tx.MsgTx().Expiry
+			if !standalone.InsideTSpendWindow(nextBlockHeight,
+				exp, g.chainParams.TreasuryVoteInterval,
+				g.chainParams.TreasuryVoteIntervalMultiplier) {
+
+				s, _ := standalone.CalculateTSpendWindowStart(exp,
+					g.chainParams.TreasuryVoteInterval,
+					g.chainParams.TreasuryVoteIntervalMultiplier)
+				log.Tracef("Skipping tspend %v because it is outside of the "+
+					"window: height %v start %v expiry %v", tx.Hash(),
+					nextBlockHeight, s, exp)
+				continue
+			}
+
+			// Ensure there are enough votes. Send in a fake block
+			// with the proper height.
+			err = g.chain.CheckTSpendHasVotes(dcrutil.NewBlock(&wire.MsgBlock{
+				Header: wire.BlockHeader{
+					Height: uint32(nextBlockHeight),
+				},
+			}), nil, dcrutil.NewTx(tx.MsgTx()))
+			if err != nil {
+				log.Tracef("Skipping tspend %v because it doesn't have enough "+
+					"votes: height %v", tx.Hash(), nextBlockHeight)
```

Comment:
It would be useful to include the reason for not including the tspend here so a simnet user can see the voting progress (or if there's some other error).

```suggestion
				log.Tracef("Skipping tspend %v because it doesn't have enough "+
					"votes: height %v reason '%v'", tx.Hash(), nextBlockHeight, err)
```

Reply:
Fixed.